prime-router/src/test/kotlin/fhirengine/engine/fhirRouterTests/RoutingTests.kt

```diff
@@ -898,18 +906,50 @@ class RoutingTests {
 
         val engine = spyk(makeFhirEngine(metadata, settings, TaskAction.route) as FHIRRouter)
 
-        every { engine.evaluateFilterConditionAsAnd(any(), any(), true, any(), any()) } returns Pair(true, null)
+        every {
+            engine.evaluateFilterConditionAsAnd(any(), any(), true, any(), any())
+        } returns Pair(true, null)
         every {
             engine.evaluateFilterConditionAsAnd(any(), any(), false, any(), any())
         } returns Pair(false, filter.toString())
 
+        every {
+            engine.evaluateFilterConditionAsOr(any(), any(), true, any(), any())
+        } returns Pair(true, null)
+        every {
+            engine.evaluateFilterConditionAsOr(any(), any(), false, any(), any())
+        } returns Pair(false, filter.toString())
+
         engine.evaluateFilterAndLogResult(filter, bundle, report, receiver, type, true)
         verify(exactly = 0) {
-            engine.logFilterResults(any(), any(), any(), any(), any())
+            engine.logFilterResults(any(), any(), any(), any(), any(), any())
         }
         engine.evaluateFilterAndLogResult(filter, bundle, report, receiver, type, false)
         verify(exactly = 1) {
-            engine.logFilterResults(any(), any(), any(), any(), any())
+            engine.logFilterResults(any(), any(), any(), any(), any(), any())
+        }
+
+        // use case for condition filter
+        val observation = FhirPathUtils.evaluate(
+            CustomContext(bundle, bundle, emptyMap<String, String>().toMutableMap()),
+            bundle,
+            bundle,
+            "Bundle.entry.resource.ofType(DiagnosticReport).result.resolve()"
+        ).first()
+
+        engine.evaluateFilterAndLogResult(
+            filter,
+            bundle,
+            report,
+            receiver,
+            ReportStreamFilterType.CONDITION_FILTER,
+            defaultResponse = true,
+            reverseFilter = false,
+            focusResource = observation,
+            useOr = true
+        )
+        verify(exactly = 1) {
```

Comment:
This isn't actually testing your new functionality? We need unit tests that test that different paths within `logFilterResults`

Reply:
Yeah that's true. I just added a new test to cover the new logging functionality